website/docs/learn-graphql/guides/thinking-in-graphql.md

```diff
@@ -0,0 +1,147 @@
+
+:::note
+
+### Broad-Query
+
+This concept might seem foreign even to those already familiar with GraphQL. To solve this at Microsoft, we had to go as far as invent a new name for this very core concept: **Broad-Query**.
+
+However, because in GraphQL _all_ queries are meant to be “broad”, we will **not** keep repeating the “Broad-Query” term. After all, we want you to walk away from this guide as someone who truly understands GraphQL!
+:::
+
+### 👍 Design from front-end perspective
+
+When designing the schema in a vacuum, it might be hard to imagine what those connections should be. However, when considered from the perspective of a concrete piece of UI, and working your way backwards, it actually becomes a lot easier.
+
+Let's consider the conversation list UI example again:
+
+<table>
+<tr>
+<th>Conversation list</th>
+<th>UI components</th>
+<th>GraphQL query</th>
+</tr>
+<tr>
+<td>
+<img src={require("./images/SmallChatList.png").default} border="1" />
+</td>
+<td>
+
+```jsx
+<ChatList>
+  <ChatListItem>
+    <Avatar />
+    <Title />
+    <LastMessage />
+    <Timestamp />
+  </ChatListItem>
+</ChatList>
+```
+
+</td>
+<td>
+
+```graphql
+query {
+  conversations {
+    participants {
+      avatarUrl
+    }
+    title
+    lastMessage
+    receivedAt
+  }
+}
+```
+
+</td>
+</tr>
+</table>
+
+The UI components were probably very natural to imagine, right? Well, as you can see, the GraphQL query you would want to be able to write is an equally natural extrapolation.
+
+Finally, completing our top-down approach from UI design to GraphQL schema, the schema to power this would need to look like this:
+
+```graphql
+type Query {
+  conversations: [Conversation]
+}
+
+type Conversation {
+  participants: [Person]
+  title: String
+  lastMessage: String
+  receivedAt: String
+}
+
+type Person {
+  avatarURL: String
+}
+```
+
+### 👎 Design from back-end perspective
+
+To contrast, let’s look at a back-end perspective schema, and how it makes it impossible to fetch all data in a single request.
+
+```graphql
+type Query {
+  conversations: [Conversation]
+  person(id: ID): Person
+}
+
+type Conversation {
+  participantIDs: [ID]
+}
+```
+
+In this case, every root-field maps to a back-end service, and it of course does not return the full data for each related entity in its response payload, but rather contains foreign-keys to those related entities.
+
+Because we can only get the IDs of participants in a conversation, rather than the actual `Person` objects they refer to, we are being forced to make an aditional request for _each_ participant in all of the conversations in the list. This is the N+1 problem and forces the UI to perform a waterfall of requests. This in turn will lead to a slow loading experience or staggered UI rendering.
+
+TODO: Expand on the need for a single-source of truth, and refer to the not yet existant guide about how a graphql client works, combined with a FLUX store.
```

Comment:
Tagging the TODO, not sure if that is to check in, or replace before hand..

Reply:
Added a new section on this: https://microsoft.github.io/graphitation/docs/learn-graphql/guides/thinking-in-graphql#generic-and-domain-specific

/cc @freiksenet Who gave me this TODO.